Theme: Events and Event types

sethjeffery opened the pull request:
Resolves #188 and #352

So now the events list, events editor, event types list, event types editor, all have up-to-date checkboxes, and better page UI.

A lot of the 'big' changes in this PR come from:

- AddEvent.php is no longer used, the entire file is discarded, and instead EventEditor.php is used both for editing and creating ([see here](https://github.com/ChurchCRM/CRM/pull/391/files#diff-09dfbdafd62bbe2b4b252e7aa3ec3c8bR431)), which saves on code.
- EditEventTypes.php had very poor markup before, so a lot of code was shifted around, indented etc.
- Whitespace. Add `?w=1` to the Github URL to ignore it.
https://github.com/ChurchCRM/CRM/pull/391/files?w=1

![](https://cloud.githubusercontent.com/assets/1628558/13213541/96a2a6f0-d941-11e5-8aad-bc6183e08c31.png)

![](https://cloud.githubusercontent.com/assets/1628558/13213506/6326e6ba-d941-11e5-94af-cb7d8cd4e89f.png)

crossan007 commented:
This looks a lot better than it did before.  My only comment is that we'd like to standardize on the DataTables plugin to display any tabular data.

https://almsaeedstudio.com/themes/AdminLTE/pages/tables/data.html

https://datatables.net/examples/styling/bootstrap.html

sethjeffery commented:
Yeah no problem with that for the list view. Actually I am not happy at all with using tables for layout in the form view. If we're ok with this kind of change, I'd like to pull it out and use `row` and `col-` instead, while using DataTables for the main list view.

crossan007 commented:
That sounds good to me - I believe that's what we're doing in Menu.php, and what I've done in all my changes so far.

dschwen commented on churchinfo/EditEventTypes.php:
```diff
@@ -47,39 +42,39 @@ function confirmDeleteOpp( Opp ) {
-  break;
+} else {
+  switch ($_POST['Action']){
+  case "ADD":
```
If you start adding 2space indented files now this will screw up our ability to do a simple project wide search/replaces of 4spaces/1tab -> 2spaces

sethjeffery replied:
A project-wide find/replace of 4 spaces to 2 spaces surely wouldn't work. There are plenty of places already where 2 spaces are used. Just look at [some of this HTML code in AddEvent.php](https://github.com/ChurchCRM/CRM/pull/391/files#diff-ae2c61854a3229dd57c59b05d859f37bL153), it's clearly 2-space indented.

dschwen replied:
Rats! :-(

DawoudIO commented:
i was about to test but looks like the branch has the broken images, so it need to be rebased or develop merged into it.

DawoudIO commented on churchinfo/mysql/upgrade/rebuild_nav_menus.sql:
```diff
@@ -63,7 +63,7 @@ INSERT INTO `menuconfig_mcf` (`mid`, `name`, `parent`, `ismenu`, `content_englis
 (50, 'events', 'root', 1, 'Events', 'Events', '', 'Events', 'bAll', NULL, 0, 0, NULL, 1, 5, 'fa-ticket'),
 (51, 'listevent', 'events', 0, 'List Church Events', 'List Church Events', 'ListEvents.php', 'List Church Events', 'bAll', NULL, 0, 0, NULL, 1, 1, NULL),
-(52, 'addevent', 'events', 0, 'Add Church Event', 'Add Church Event', 'EventNames.php', 'Add Church Event', 'bAll', NULL, 0, 0, NULL, 1, 2, NULL),
+(52, 'addevent', 'events', 0, 'Add Church Event', 'Add Church Event', 'EventEditor.php', 'Add Church Event', 'bAll', NULL, 0, 0, NULL, 1, 2, NULL),
```
make sure you make that change in the install.sql also

sethjeffery replied:
I'll update `install.sql`. 53 ("List Event Types") is still useful, it's just that both 52 and 53 were going to the same URL which was pointless. They should both exist but go to different pages.

DawoudIO commented:
once you address the comments please merge

sethjeffery commented:
Will do once I get a free evening again @DawoudIO (though my preference is not to merge my own PRs!)

DawoudIO commented:
Let me know and I'll merge it

sethjeffery commented:
This is now ready for review. I added DataTables support to both Events and Event Types list, and I performed some major cleanup of the way the code works. I also cleaned up the tables by making the actions list simpler using icons and tooltips.

![image](https://cloud.githubusercontent.com/assets/1628558/13410477/e609fff6-df2f-11e5-85b5-fed3d30b2eea.png)


In particular, I never liked the idea that clicking "New Event Type" would immediately create an item in the database before you entered any information, meaning that you couldn't cancel. Also, having the form directly in the table was incompatible with DataTables.. so I stripped it out completely and used a proper bootstrap form that you can cancel if you want.

It also looks nice on mobile.

![](https://gyazo.com/8aeed1f63616db843869e8aab36442fb.gif)

crossan007 commented:
When called with nothing in the POST body, ListEvents.php displays no events while leading you to believe that "All Events" are displayed upon load:
![image](https://cloud.githubusercontent.com/assets/11679900/13414381/6f45b720-df20-11e5-8a17-dcfd7099193b.png)


When you select a filter, it re-loads with POST body data, and displays events:
![image](https://cloud.githubusercontent.com/assets/11679900/13414402/91f1abf8-df20-11e5-9c59-ba75f1ffdb39.png)

When you select the "All" filter (The same as it was on initial page load), All events are displayed:
![image](https://cloud.githubusercontent.com/assets/11679900/13414414/a9e93136-df20-11e5-8518-29278f7ec6fc.png)

DawoudIO commented:
Also checkout https://almsaeedstudio.com/themes/AdminLTE/pages/calendar.html as it is part of our skin

DawoudIO commented:
@sethjeffery  is this ready for a merge

sethjeffery commented:
No not yet. I'll rebase when it's ready. I've just had no free time to move the files around.

Have we decided whether it's right to move *all** js into "skin"? (Whether it's UI-related or not?)

DawoudIO commented:
i'm open to moving all HTML related files to skin.

crossan007 commented on churchinfo/EventNames.php:
```diff
@@ -168,76 +148,64 @@ function confirmDeleteOpp( Opp ) {
+          <div class='col-sm-6'>
+            <div class='row form-radio-list'>
+              <div class='col-xs-12'>
+                <?php $render->Radio('None', 'newEvtTypeRecur', 'none'); ?>
```
Can you add some jQuery to this so that selecting "None" for recurrance pattern hides the recurrance pattern selection area?   It may also be that the recurrance pattern would dictate a different set of "pattern" definitions.

![image](https://cloud.githubusercontent.com/assets/11679900/13798125/8925f54a-eae9-11e5-9b39-6892e108a65a.png)

sethjeffery replied:
Done.
![](https://gyazo.com/dccb4341865d16bd44768b5b61a62ba4.gif)

crossan007 commented:
It looks like confirmDeleteOpp is not used at https://github.com/ChurchCRM/CRM/blob/ui_events/churchinfo/EventNames.php#L39

crossan007 commented:
ListEvents.php content is not contained in a box div (does not have a white background around the actual content of the page)

crossan007 commented:
EventEditor.php "Return to Events" button is outside of the main content div:

![image](https://cloud.githubusercontent.com/assets/11679900/13798279/11186f40-eaeb-11e5-8f95-40daa5173bff.png)

crossan007 commented:
EventEditor.php spacing between Attendance Notes, and the text box:

![image](https://cloud.githubusercontent.com/assets/11679900/13798299/3a895434-eaeb-11e5-9a01-cf2f0ed5e3b0.png)

crossan007 commented:
Checkin.php is not themed:

![image](https://cloud.githubusercontent.com/assets/11679900/13798351/e5ad78e0-eaeb-11e5-896a-5da617ef7552.png)


We can track this separately if you want.

crossan007 commented:
Adding a new event type without any explicit counts in the text box for CSV entry causes ", Total" to be displayed in the Attendance Counts column on the Event Types Data Table:

crossan007 commented:
Creating a new event with no recurrence pattern causes the event title to start with a "-"

![image](https://cloud.githubusercontent.com/assets/11679900/13798459/159ca4f8-eaed-11e5-9a3e-5d8fd66f181e.png)

sethjeffery commented:
Ouch. :)

sethjeffery commented:
In regards to elements inside and outside of box, I'd really appreciate if we defined the UX rules as to how boxes are used, what their purpose is, and what elements are allowed to be outside.

For example, I purposely put the back button outside of the box because I felt that navigation shouldn't be part of the content, but perhaps it should?

sethjeffery commented:
> Creating a new event with no recurrence pattern causes the event title to start with a "-"

I can't reproduce this :disappointed:

sethjeffery commented:
> Adding a new event type without any explicit counts in the text box for CSV entry causes ", Total" 

Fixed.

sethjeffery commented:
Ok, ready for next review. @crossan007
